pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/BasePartitionUpsertMetadataManager.java

```diff
@@ -847,6 +847,8 @@ public void takeSnapshot() {
       long duration = System.currentTimeMillis() - startTime;
       _serverMetrics.addTimedTableValue(_tableNameWithType, ServerTimer.UPSERT_SNAPSHOT_TIME_MS, duration,
           TimeUnit.MILLISECONDS);
+    } catch (Exception e) {
+      _logger.warn("Error encountered while taking snapshot", e);
```

Comment:
nit: "Caught exception while ..." to be a bit consistent with other logs for exceptions.